hedera-node/src/main/java/com/hedera/services/txns/consensus/SubmitMessageTransitionLogic.java

```diff
@@ -75,55 +68,38 @@ public SubmitMessageTransitionLogic(
 	public void doStateTransition() {
 		var transactionBody = transactionContext.accessor().getTxn();
 		var op = transactionBody.getConsensusSubmitMessage();
-
-		if (op.getMessage().isEmpty()) {
-			transactionContext.setStatus(INVALID_TOPIC_MESSAGE);
-			return;
-		}
-
-		if(op.getMessage().size() > globalDynamicProperties.messageMaxBytesAllowed() ) {
-			transactionContext.setStatus(MESSAGE_SIZE_TOO_LARGE);
-			return;
-		}
-
-		var topicStatus = validator.queryableTopicStatus(op.getTopicID(), topics.get());
-		if (OK != topicStatus) {
-			transactionContext.setStatus(topicStatus);
-			return;
-		}
-
+		/* --- Extract from gRPC ---*/
+		final var message = op.getMessage();
+		final var topicId = Id.fromGrpcTopic(op.getTopicID());
+
+		/* --- Validate ---*/
+		validateFalse(message.isEmpty(), INVALID_TOPIC_MESSAGE);
+		validateFalse(message.size() > globalDynamicProperties.messageMaxBytesAllowed(), MESSAGE_SIZE_TOO_LARGE);
+		/* --- load the model here, so we can maintain the order of validations --- */
+		final var topic = topicStore.loadTopic(topicId);
 		if (op.hasChunkInfo()) {
 			var chunkInfo = op.getChunkInfo();
-			if (!(1 <= chunkInfo.getNumber() && chunkInfo.getNumber() <= chunkInfo.getTotal())) {
-				transactionContext.setStatus(INVALID_CHUNK_NUMBER);
-				return;
-			}
-			if (!chunkInfo.getInitialTransactionID().getAccountID().equals(
-					transactionBody.getTransactionID().getAccountID())) {
-				transactionContext.setStatus(INVALID_CHUNK_TRANSACTION_ID);
-				return;
-			}
-			if (1 == chunkInfo.getNumber() &&
-					!chunkInfo.getInitialTransactionID().equals(transactionBody.getTransactionID())) {
-				transactionContext.setStatus(INVALID_CHUNK_TRANSACTION_ID);
-				return;
-			}
+			validateFalse(!(1 <= chunkInfo.getNumber() && chunkInfo.getNumber() <= chunkInfo.getTotal()), INVALID_CHUNK_NUMBER);
+			validateFalse(!chunkInfo.getInitialTransactionID().getAccountID().equals(
+					transactionBody.getTransactionID().getAccountID()), INVALID_CHUNK_TRANSACTION_ID);
+			validateFalse(1 == chunkInfo.getNumber() &&
+					!chunkInfo.getInitialTransactionID().equals(transactionBody.getTransactionID()), INVALID_CHUNK_TRANSACTION_ID);
 		}
 
-		var topicId = EntityNum.fromTopicId(op.getTopicID());
-		var mutableTopic = topics.get().getForModify(topicId);
-		try {
-			mutableTopic.updateRunningHashAndSequenceNumber(
-					transactionBody.getTransactionID().getAccountID(),
-					op.getMessage().toByteArray(),
-					op.getTopicID(),
-					transactionContext.consensusTime());
-			transactionContext.setTopicRunningHash(mutableTopic.getRunningHash(), mutableTopic.getSequenceNumber());
-			transactionContext.setStatus(SUCCESS);
-		} catch (IOException e) {
-			log.error("Updating topic running hash failed.", e);
-			transactionContext.setStatus(INVALID_TRANSACTION);
-		}
+		/* --- Do the business logic --- */
+		final var payer = Id.fromGrpcAccount(transactionBody.getTransactionID().getAccountID());
```

Comment:
Since we are loading the GRPC object and translating it to a model object, we can place this line of code at the top in the first section of the transition logic